src/network/networkd-link.c

```diff
@@ -121,7 +121,7 @@ bool link_ipv4ll_enabled(Link *link) {
         if (!link->network)
                 return false;
 
-        if (STRPTR_IN_SET(link->kind, "vrf", "wireguard"))
+        if (STRPTR_IN_SET(link->kind, "vrf", "wireguard", "sit"))
```

Comment:
This should be for all the tunnel ipip/gre/gre6/vti/vti6 etc. Please define a macro ```dev_is_tunnel``` something like this and call that here.
```
sit-test: Gained carrier
sit-test: Acquiring IPv4 link-local address
Assertion '!ether_addr_is_null(&ll->mac)' failed at ../src/libsystemd-network/sd-ipv4ll.c:228, function ipv4ll_start_internal(). Ignoring.
sit-test: Could not acquire IPv4 link-local address: Invalid argument

```